tasks/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -0,0 +1,104 @@
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Task to create internalrequests to add contributions to index images
+  params:
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+      default: "snapshot_spec.json"
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+      default: "data.json"
+    - name: binaryImage
+      type: string
+      description: binaryImage value updated by update-ocp-tag task
+    - name: fromIndex
+      type: string
+      description: fromIndex value updated by update-ocp-tag task
+    - name: requestTimeout
+      type: string
+      default: "180"
+      description: InternalRequest timeout
+  results:
+    - name: fbcOptIn
+      description: Result to set weather the contribution is FBC opt in
+    - name: requestResultsFile
+      description: Internal Request results file
+    - name: requestMessage
+      description: Internal Request message
+    - name: requestReason
+      description: Internal Request reason
+  workspaces:
+    - name: data
+      description: workspace to read and save files
+  steps:
+    - name: add-contribution
+      image: quay.io/hacbs-release/release-utils:a9603c0ae1982be5e1142d44765b1475adc7ad40
+      script: |
+        #!/usr/bin/env sh
+        #
+        set -e
+
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        # Set the value for requestResultsFile
+        echo "$(workspaces.data.path)/ir-$(context.taskRun.uid)-result.txt" > $(results.requestResultsFile.path)
+
+        # IIB is the default internal pipeline for fbc requests
+        request=$(jq -r '.fbc.request // "iib"' ${DATA_FILE})
+
+        default_build_timeout_seconds="300"
+        iib_overwrite_from_index_credential=$(jq -r \
+          '.fbc.iibOverwriteFromIndexCredential // "iib-overwrite-fromimage-credentials"' ${DATA_FILE})
+        iib_service_config_secret=$(jq -r '.fbc.iibServiceConfigSecret // "iib-services-config"' ${DATA_FILE})
+        iib_service_account_secret=$(jq -r '.fbc.iibServiceAccountSecret // "iib-service-account"' ${DATA_FILE})
+        build_tags=$(jq -r '.fbc.buildTags // "[]"' ${DATA_FILE})
+        add_arches=$(jq -r '.fbc.addArches // "[]"' ${DATA_FILE})
+        build_timeout_seconds=$(jq -r --arg build_timeout_seconds ${default_build_timeout_seconds} \
+            '.fbc.buildTimeoutSeconds // $build_timeout_seconds' ${DATA_FILE})
+        target_index=$(jq -r '.fbc.targetIndex' ${DATA_FILE})
+        fbc_fragment=$(jq -cr '.components[0].containerImage' ${SNAPSHOT_PATH} |tr -d "\n")
```

Comment:
Why the `tr`? `containerImage` should be a string without new lines, no?

Reply:
Indeed. There was a bug (already fixed) in collectd-data that was adding the `\n` to the `fbcFragment` result. Removed as we won´t use this result anymore.